Source/Diagnostics/ComputeDiagFunctors/BackTransformParticleFunctor.cpp

```diff
@@ -145,6 +145,7 @@ BackTransformParticleFunctor::operator () (PinnedMemoryParticleContainer& pc_dst
                    if (Flag[i] == 1) GetParticleLorentzTransform(dst_data, src_data, i,
                                                                  old_size + IndexLocation[i]);
                 });
+                amrex::Gpu::synchronize();
```

Comment:
Awesome catch.

Can you add a comment why we need to synchronize here? Is this because we reuse `FlagForPartCopy` and `IndexForPartCopy` for every particle tile in the loop?

Reply:
thats right!

Reply:
oh but this is merged already. I will add a mini-PR